Softeq.XToolkit.Common/Collections/ObservableRangeCollection.cs

```diff
@@ -211,22 +211,19 @@ public void InsertRange(IEnumerable<T> collection, int startIndex)
                 return;
             }
 
-            var changedItems = collection is List<T> ? (List<T>) collection : new List<T>(collection);
-            var index = Items.IndexOf(changedItems[0]);
-
-            for (var i = 0; i < changedItems.Count; i++)
+            var removedItems = new List<T>();
+            foreach (var item in collection)
             {
-                if (!Items.Remove(changedItems[i]))
+                if (Items.Remove(item))
                 {
-                    changedItems.RemoveAt(i); // Can't use a foreach because changedItems is intended to be (carefully) modified
-                    i--;
+                    removedItems.Add(item);
                 }
             }
 
             OnPropertyChanged(EventArgsCache.CountPropertyChanged);
             OnPropertyChanged(EventArgsCache.IndexerPropertyChanged);
             OnCollectionChanged(
-                new NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Remove, changedItems, index));
+                new NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Remove, removedItems));
```

Comment:
What about `startingIndex` of the event args?

Reply:
I was worried about it as well, but then I read the note in documentation to this method:
"NOTE: with notificationMode = Remove, removed items starting index is not set because items are not guaranteed to be consecutive."
Looks like this index was not valid anyways, so I've just removed it)

Reply:
Make sense, but currently, Bindings Collections used OldStartingIndex to handle collection changes.

The default value is `-1` does it affects our collection handlers for iOS and Android?